### PR TITLE
Incoperate 'Dependencies.toml' to get updated with the build

### PR DESCRIPTION
**Pull request by pramodya1994**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/28562

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by hevayo — APPROVED** [review]

**Review by hevayo — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java [hunk 1]
```diff
@@ -202,10 +207,16 @@ private Path emitBala(Path filePath) {
 
         List<PlatformLibrary> platformLibraries = new ArrayList<>();
         for (Map<String, Object> dependency : javaPlatform.dependencies()) {
-            String dependencyFilePath = (String) dependency.get(JarLibrary.KEY_PATH);
             String artifactId = (String) dependency.get(JarLibrary.KEY_ARTIFACT_ID);
             String version = (String) dependency.get(JarLibrary.KEY_VERSION);
             String groupId = (String) dependency.get(JarLibrary.KEY_GROUP_ID);
+
+            String dependencyFilePath = (String) dependency.get(JarLibrary.KEY_PATH);
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java:214]: Do we have tests to test this path ?

**Comment by nadeeshaan** [comment]
Does this get updated in the offline build mode as well? In the tooling (language server) scenarios, we build the project via the project API and does this also affect the particular path as well?

**Comment by pramodya1994** [comment]
> Does this get updated in the offline build mode as well? In the tooling (language server) scenarios, we build the project via the project API and does this also affect the particular path as well?

We have introduced `project.save()` to the project API which writes dependencies to the `Dependencies.toml`

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java (hunk 1 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java:214]: There's an integration test called `MavenTestCase` covering this scenario and unit tests in `MavenResolverTest` class. These unit tests are testing `Utils.getJarPath` functions which have used inside `getPlatformLibPath` method in `JBallerinaBackend`.

**Pull request merged**